nifi-nar-bundles/nifi-pgp-bundle/nifi-pgp-processors/src/test/java/org/apache/nifi/processors/pgp/DecryptContentPGPTest.java

```diff
@@ -0,0 +1,391 @@
+
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.CoreMatchers.isA;
+import static org.junit.Assert.assertTrue;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DecryptContentPGPTest {
+    private static final int ENCRYPTION_ALGORITHM = SymmetricKeyAlgorithmTags.AES_256;
+
+    private static final boolean INTEGRITY_ENABLED = true;
+
+    private static final boolean INTEGRITY_DISABLED = false;
+
+    private static final String PASSPHRASE = UUID.randomUUID().toString();
+
+    private static final String FILE_NAME = String.class.getSimpleName();
+
+    private static final char FILE_TYPE = PGPLiteralDataGenerator.TEXT;
+
+    private static final long MODIFIED_MILLISECONDS = 86400000;
+
+    private static final Date MODIFIED = new Date(MODIFIED_MILLISECONDS);
+
+    private static final String DATA = String.class.getName();
+
+    private static final Charset DATA_CHARSET = StandardCharsets.UTF_8;
+
+    private static final int BUFFER_SIZE = 128;
+
+    private static final boolean NESTED_SIGNATURE_DISABLED = false;
+
+    private static final String SERVICE_ID = PGPPrivateKeyService.class.getSimpleName();
+
+    private static PGPSecretKey rsaSecretKey;
+
+    private static PGPPrivateKey rsaPrivateKey;
+
+    private static PGPPublicKey elGamalPublicKey;
+
+    private static PGPPrivateKey elGamalPrivateKey;
+
+    private TestRunner runner;
+
+    @Mock
+    private PGPPrivateKeyService privateKeyService;
+
+    @BeforeClass
+    public static void setKeys() throws Exception {
+        rsaSecretKey = PGPSecretKeyGenerator.generateRsaSecretKey(PASSPHRASE.toCharArray());
+
+        final PBESecretKeyDecryptor decryptor = new JcePBESecretKeyDecryptorBuilder().build(PASSPHRASE.toCharArray());
+        rsaPrivateKey = rsaSecretKey.extractPrivateKey(decryptor);
+        final PGPSecretKeyRing dsaElGamalSecretKeyRing = PGPSecretKeyGenerator.generateDsaElGamalSecretKeyRing(PASSPHRASE.toCharArray());
+        for (final PGPSecretKey secretKey : dsaElGamalSecretKeyRing) {
+            final PGPPublicKey publicKey = secretKey.getPublicKey();
+            if (PGPPublicKey.ELGAMAL_ENCRYPT == publicKey.getAlgorithm()) {
+                elGamalPrivateKey = secretKey.extractPrivateKey(decryptor);
+                elGamalPublicKey = publicKey;
+            }
+        }
+    }
+
+    @Before
+    public void setRunner() {
+        runner = TestRunners.newTestRunner(new DecryptContentPGP());
+    }
+
+    @Test
+    public void testMissingProperties() {
+        runner.assertNotValid();
+    }
+
+    @Test
+    public void testFailureEncryptedDataNotFound() {
+        runner.setProperty(DecryptContentPGP.PASSPHRASE, PASSPHRASE);
+        runner.enqueue(new byte[]{});
+        runner.run();
+
+        assertFailureExceptionLogged(PGPProcessException.class);
+    }
+
```

Comment:
Should it be possible to configure both Password-based encryption and Private Key Encryption at the same time in this processor?  Either way, it would be great to see a unit test showing what happens when you configure the processor both ways.

Reply:
Great question! RFC 4880 Section 5.3 describes this as a valid scenario, so I will include additional unit tests.